src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/Utils.java

```diff
@@ -2831,6 +2834,26 @@ public DocCommentTree getDocCommentTree(Element element) {
         return getBlockTags(element, USES);
     }
 
+    public DocTree getPreviewTree(Element element) {
+        return getBody(element).stream()
+                               .filter(t -> t.getKind() == Kind.UNKNOWN_INLINE_TAG)
+                               .filter(t -> "preview".equals(((UnknownInlineTagTree) t).getTagName()))
+                               .findAny()
+                               .orElse(null);
+    }
+
+    public String getPreviewTreeSummaryOrDetails(DocTree t, boolean summary) {
+        UnknownInlineTagTree previewTag = (UnknownInlineTagTree) t;
+        List<? extends DocTree> previewContent = previewTag.getContent();
+        String previewText = ((TextTree) previewContent.get(0)).getBody();
```

Comment:
This looks unreasonably fragile. And, I thought the tag was going away... at least according to earlier files in this review.

Reply:
This was intended to allow Preview APIs to provide their own text instead of the generic one. But, looking again at JEP 12, this shouldn't be supported, so removing.